src/checks/incorrect_delimiter.rs

```diff
@@ -24,8 +24,21 @@ impl Default for IncorrectDelimiterChecker<'_> {
 impl Check for IncorrectDelimiterChecker<'_> {
     fn run(&mut self, line: &LineEntry) -> Option<Warning> {
         let key = line.get_key()?;
-        if key.trim().chars().any(|c| !c.is_alphanumeric() && c != '_') {
-            return Some(Warning::new(line.clone(), self.name(), self.message(&key)));
+
+        // delimiters occur /between/ characters, not as the initial character, so we should
+        // remove all invalid leading characters before checking for incorrect delimiters
+        let cleaned_key = remove_all_invalid_leading_chars(&key);
```

Comment:
@baile320 Please move these changes to another PR.
We should merge these changes before merging the implemenation of `LeadingCharacterFixer`.

Reply:
I moved this logic to a separate PR: #267 . I can update this PR when #267 is merged to master, since it looks like the module file structured changed a little bit.